components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java

```diff
@@ -206,17 +246,180 @@ public void testHandleClaimsForOAuthAuthzReqMessageContext() throws Exception {
+                thenReturn(authorizationGrantCacheEntry);
+
+        AuthenticatedUser user = mock(AuthenticatedUser.class);
+        when(requestMsgCtx.getAuthorizedUser()).thenReturn(user);
+        when(user.isFederatedUser()).thenReturn(false);
+
+        OAuth2AccessTokenReqDTO oAuth2AccessTokenReqDTO = mock(OAuth2AccessTokenReqDTO.class);
+        when(requestMsgCtx.getOauth2AccessTokenReqDTO()).thenReturn(oAuth2AccessTokenReqDTO);
+        when(oAuth2AccessTokenReqDTO.getTenantDomain()).thenReturn(SAMPLE_TENANT_DOMAIN);
+        when(oAuth2AccessTokenReqDTO.getClientId()).thenReturn(SAMPLE_CLIENT_ID);
+
+        mockStatic(OAuth2ServiceComponentHolder.class);
+        applicationManagementService = mock(ApplicationManagementService.class);
+        when(OAuth2ServiceComponentHolder.getApplicationMgtService()).thenReturn(applicationManagementService);
+        when(applicationManagementService.getServiceProviderNameByClientId(anyString(), anyString(), anyString()))
+                .thenReturn(SAMPLE_SERVICE_PROVIDER);
+
+        startTenantFlowProcess();
+
+        samlAssertionClaimsCallback.handleCustomClaims(jwtClaimsSet, requestMsgCtx);
+        assertEquals(jwtClaimsSet.getAllClaims().size(), 8, "Claims are not successfully set.");
+    }
+
+    @Test
+    public void testHandleClaimsForOAuthAuthzReqMessageContextNullAccessToken() throws Exception {
+        jwtClaimsSet = new JWTClaimsSet();
+        oAuthAuthzReqMessageContext = mock(OAuthAuthzReqMessageContext.class);
+        String[] approvedScopes = {"openid", "testScope1", "testScope2"};
+        when(oAuthAuthzReqMessageContext.getApprovedScope()).thenReturn(approvedScopes);
+        when(oAuthAuthzReqMessageContext.getProperty(OAuthConstants.ACCESS_TOKEN))
+                .thenReturn(null);
+
+        OAuth2AuthorizeReqDTO oAuth2AuthorizeReqDTO = mock(OAuth2AuthorizeReqDTO.class);
+        when(oAuthAuthzReqMessageContext.getAuthorizationReqDTO()).thenReturn(oAuth2AuthorizeReqDTO);
+
+        AuthenticatedUser authenticatedUser = mock(AuthenticatedUser.class);
+        when(oAuth2AuthorizeReqDTO.getUser()).thenReturn(authenticatedUser);
+        when(authenticatedUser.isFederatedUser()).thenReturn(false);
+
+        when(authenticatedUser.getUserName()).thenReturn("user1234");
+        when(authenticatedUser.getUserStoreDomain()).thenReturn("abc.com");
+
+
+        ApplicationManagementService mockApplicationManagementService = mock(ApplicationManagementService.class);
+        mockStatic(OAuth2ServiceComponentHolder.class);
+        when(OAuth2ServiceComponentHolder.getApplicationMgtService()).thenReturn(mockApplicationManagementService);
+
+        when(mockApplicationManagementService.getServiceProviderNameByClientId(anyString(), anyString(), anyString()))
+                .thenReturn(SAMPLE_SERVICE_PROVIDER);
+        ServiceProvider serviceProvider = mock(ServiceProvider.class);
```

Comment:
Why do you want to mock service provider ? Instead can't u create a new instance of SP ?